### PR TITLE
Notification email on admin registration approval

### PR DESCRIPTION
**Pull request by carlitorweb**
Pull Request for Issue #13620 

### Summary of Changes
Sending a notification email, after an Admin activate a user/users, on the user's list of com_users 

### Testing Instructions

- Register as a new user
- Go to the Users list inside the administration, and activate the user/users
- Check the email is sent to the user, notifying him that it has been approved on the site
----------------------------------------------------------------------
- Edit a user that is not activated
- Press the button you will find under the Account Details tab
- Check the user is activated and an email is sent for notifying him that it has been approved on the site
----------------------------------------------------------------------
- Edit a user that is activated, but that has not yet entered in the site for the first time
- Check the same button is there under the Account Details tab, and press it
- Check an email is sent for notifying him that it has been approved on the site

### Expected result
An email is sent to the user, notifying him that it has been approved on the site

### Actual result
The user is activated but is not notified of this change

### Request for feedback (SOLVED)
This change also requires that within the user edit view, a button appears that allows an administrator to activate the user from there. 

For this, I added a new radio field "activation" for handle this under Accounts Details, but this field store a hash when the user It is registered in the site then this new field is not convenient.
new field - edit view

I can create a new field under the params fields, and handle this as a param, but I not think this need be under the Basic Settings tab.

So, my doubt is, What is the best way to treat this?

### Documentation Changes Required
- New com_users.user button added. 
- Now when you activate a user on com_users list, send an email

### CODE REVIEW

**Comment by brianteeman** [comment]
why not a simple button that "activates" the user and "sends the email" - something like here (ignore the extra tabs)

chrome_2018-05-02_14-27-42

**Comment by carlitorweb** [comment]
That is a good point @brianteeman . Thank you. I will work on that way.

**Comment by brianteeman** [comment]
@carlitorweb you might find the second half of this blog post (and the comments) useful https://brian.teeman.net/joomla/870-authorising-and-approving-joomla-users

Thanks for working on this

**Comment by carlitorweb** [comment]
Really have good comments that blog post. Carlos found an easy and good way for this [https://github.com/carcam/plg_user_sendactivationmail/blob/master/sendactivation/sendactivationmail.php](url) . 
But that comment about _map this button with both actions, "activation" AND "resend email_ sound good too.

File: administrator/components/com_users/views/user/tmpl/edit.php [hunk 1]
```diff
@@ -63,6 +65,14 @@
 								<?php // Disables autocomplete ?> <input type="password" style="display:none">
 							<?php endif; ?>
 							<?php echo $field->input; ?>
+							<?php // If the user has not logged into the site for the first time, then either an administrator has not activated it ?>
+							<?php // Or in some way the notification mail has not been received yet ?>
+							<?php if($field->fieldname == 'name' && $this->form->getValue('lastvisitDate') === JFactory::getDbo()->getNullDate()) : ?>
```
Comment (Quy) [inline administrator/components/com_users/views/user/tmpl/edit.php]: Add space after `if`

File: administrator/components/com_users/controllers/user.php [hunk 2]
```diff
@@ -85,4 +85,52 @@ protected function postSaveHook(JModelLegacy $model, $validData = array())
...
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function activate()
+	{
+		// Check for request forgeries.
+		JSession::checkToken() or jexit(JText::_('JINVALID_TOKEN'));		
+
```
Comment (Quy) [inline administrator/components/com_users/controllers/user.php]: Remove tabs

File: administrator/components/com_users/controllers/user.php [hunk 3]
```diff
@@ -85,4 +85,52 @@ protected function postSaveHook(JModelLegacy $model, $validData = array())
...
+	 * Method to activate a user.
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function activate()
+	{
+		// Check for request forgeries.
+		JSession::checkToken() or jexit(JText::_('JINVALID_TOKEN'));
```
Comment (Quy) [inline administrator/components/com_users/controllers/user.php]: Remove trailing tabs

File: administrator/components/com_users/models/user.php [hunk 4]
```diff
@@ -538,6 +538,7 @@ public function activate(&$pks)
 	{
 		$dispatcher = JEventDispatcher::getInstance();
 		$user       = JFactory::getUser();
+		$config 	= JFactory::getConfig();
```
Comment (Quy) [inline administrator/components/com_users/models/user.php]: Align `=`

**Comment by carlitorweb** [comment]
@brianteeman what you think? (the warning: Could not start mail function, is because of a wrong config on my local)
Thank @Quy 
video_2018-05-03_105812

**Comment by brianteeman** [comment]
I'm on my phone but the button says send email? And it activates and sends email? 

If so then I personally would just call the button "activate"

To me that would be the expected behaviour

**Comment by carlitorweb** [comment]
Yes, send email, but active at same time, because if send an email saying to the user can log in now, then that user need be activated for that.

File: administrator/components/com_users/controllers/user.php (hunk 3 above)
Comment (Quy) [reply administrator/components/com_users/controllers/user.php]: Also line 99 with trailing tabs.

**Comment by carlitorweb** [comment]
Done. Thank again @Quy

File: administrator/components/com_users/models/user.php [hunk 5]
```diff
@@ -537,7 +537,8 @@ public function block(&$pks, $value = 1)
 	public function activate(&$pks)
 	{
 		$dispatcher = JEventDispatcher::getInstance();
-		$user       = JFactory::getUser();
+		$user 		= JFactory::getUser();
```
Comment (Quy) [inline administrator/components/com_users/models/user.php]: Align `=`

**Comment by brianteeman** [comment]
I still think the text on the button is a little unclear

The admin might think that they have to scroll down and click activate (like they did before) and then click on the button to send the email.

When in fact the button (correctly) does both. So I would prefer something like
"Activate and send email"

File: administrator/language/en-GB/en-GB.com_users.ini [hunk 6]
```diff
@@ -345,11 +346,14 @@ COM_USERS_USER_FIELD_USERNAME_DESC="Enter the login name (Username) for the user
 COM_USERS_USER_FIELD_USERNAME_LABEL="Login Name"
 COM_USERS_USER_GROUPS_HAVING_ACCESS="User Groups Having Viewing Access"
 COM_USERS_USER_HEADING="User"
+COM_USERS_USER_NOTIFIED="An email has been sent notifying the user that it has been activated."
 COM_USERS_USER_OTEPS="One time emergency passwords"
 COM_USERS_USER_OTEPS_DESC="If you do not have access to your two factor authentication device you can use any of the following passwords instead of a regular security code. Each one of these emergency passwords is immediately destroyed upon use. We recommend printing these passwords out and keeping the printout in a safe and accessible location, eg your wallet or a safety deposit box."
 COM_USERS_USER_OTEPS_WAIT_DESC="There are no emergency one time passwords generated in your account. The passwords will be generated automatically and displayed here as soon as you activate two factor authentication."
+COM_USERS_USER_RESEND_BUTTONTEXT="Resend a notification email"
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_users.ini]: ==> "Resend activation email"

**Comment by carlitorweb** [comment]
This commit performs this task through ajax. In this way, the user no need redirect to the com_users list. I will fix the drone errors

**Comment by infograf768** [comment]
Still many CS, but most important is that the mail does not contain the strings values:

screen shot 2018-05-06 at 09 41 05

EDIT: which is normal as they are only present in frontend... 😄 

NOTE: Once the strings added, let's not forget the language used would be the language of the admin UI when the button is clicked, not always the language used by the new user who registered on frontend.

**Comment by infograf768** [comment]
You need to add these language strings in backend.

**Comment by carlitorweb** [comment]
Yes, I on it. But I was config phpcs well here for Joomla, for start coding better for the CMS

**Comment by infograf768** [comment]
> But I was config phpcs well here for Joomla, for start coding better for the CMS

👍

**Comment by carlitorweb** [comment]
Thank you again @infograf768 . I added the 2 missing string, and also, change the strings for the 2 different activations. When you want active the user for 1st time, and when you want send a reminder

**Comment by infograf768** [comment]
Looks like the strings should be correctly grouped for Reminder and Activate in the model.

**Comment by carlitorweb** [comment]
>correctly grouped for Reminder and Activate

Sorry, not understand well what you suggest.

File: administrator/components/com_users/controllers/activation.json.php [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+	{
+		$message = JText::_('COM_USERS_USER_NOTIFIED');
+		$error = false;
+		$checkResult = $this->checkFormData($message, $error);
+
+		try
+		{
+			echo new JResponseJson($checkResult, $message, $error);
+		}
+
```
Comment (Quy) [inline administrator/components/com_users/controllers/activation.json.php]: Remove blank line.

File: administrator/components/com_users/views/user/tmpl/edit.php [hunk 8]
```diff
@@ -63,6 +123,14 @@
 								<?php // Disables autocomplete ?> <input type="password" style="display:none">
 							<?php endif; ?>
 							<?php echo $field->input; ?>
+							<?php // If the user has not logged into the site for the first time, then either an administrator has not activated it ?>
+							<?php // Or in some way the notification mail has not been received yet ?>
+							<?php if ($field->fieldname == 'name' && $this->form->getValue('lastvisitDate') === JFactory::getDbo()->getNullDate()) : ?>
+								<?php $aText = (!$this->form->getValue('block')) ? $resendText : JText::_('COM_USERS_USER_SEND_BUTTONTEXT'); ?>
```
Comment (Quy) [inline administrator/components/com_users/views/user/tmpl/edit.php]: Outside `()` not necessary.

**Comment by infograf768** [comment]
I meant using the right language constants for the mail sent:

`COM_USERS_EMAIL_ACTIVATED_BY_ADMIN_ACTIVATION_SUBJECT`
should go with
`COM_USERS_EMAIL_NOTIFICATION_ACTIVATED_BODY`


`COM_USERS_EMAIL_NOTIFICATION_ACTIVATED_REMINDER_SUBJECT`
with
`COM_USERS_EMAIL_NOTIFICATION_ACTIVATED_REMINDER_BODY`

in
```
+						if (!empty($wasActive))
+						{
+							// Then we wanted active the user and send a email notification
+							// Compute the user mail notification subject.
+							$emailSubject = JText::sprintf(
+								'COM_USERS_EMAIL_ACTIVATED_BY_ADMIN_ACTIVATION_SUBJECT',
+								$table->name,
+								$config->get('sitename')
+							);
+
+							// Compute the user mail notification body.
+							$emailBody = JText::sprintf(
+								'COM_USERS_EMAIL_NOTIFICATION_ACTIVATED_REMINDER_BODY',
+								$table->name,
+								JUri::root(),
+								$table->username
+							);
+						}
+						else
+						{
+							// Then we wanted only send again the email notification
+							$emailSubject = JText::sprintf(
+								'COM_USERS_EMAIL_NOTIFICATION_ACTIVATED_REMINDER_SUBJECT',
+								$config->get('sitename')
+							);
+
+							// Compute the user mail notification body.
+							$emailBody = JText::sprintf(
+								'COM_USERS_EMAIL_NOTIFICATION_ACTIVATED_BODY',
+								$table->name,
+								JUri::root(),
+								$table->username
+							);
+						}
```

**Comment by carlitorweb** [comment]
Good catch @infograf768 . Thank you again, and sorry for no understand well before.

**Comment by carlitorweb** [comment]
Why Travis failed here?

**Comment by carlitorweb** [comment]
Sorry here, made a mistake

File: administrator/language/en-GB/en-GB.com_users.ini [hunk 9]
```diff
@@ -345,11 +350,15 @@ COM_USERS_USER_FIELD_USERNAME_DESC="Enter the login name (Username) for the user
...
 COM_USERS_USER_HEADING="User"
+COM_USERS_USER_NOTIFIED="An email has been sent notifying the user that it has been activated."
 COM_USERS_USER_OTEPS="One time emergency passwords"
 COM_USERS_USER_OTEPS_DESC="If you do not have access to your two factor authentication device you can use any of the following passwords instead of a regular security code. Each one of these emergency passwords is immediately destroyed upon use. We recommend printing these passwords out and keeping the printout in a safe and accessible location, eg your wallet or a safety deposit box."
 COM_USERS_USER_OTEPS_WAIT_DESC="There are no emergency one time passwords generated in your account. The passwords will be generated automatically and displayed here as soon as you activate two factor authentication."
+COM_USERS_USER_RESEND_BUTTONTEXT="Activate and resend activation email"
 COM_USERS_USER_SAVE_FAILED="An error was encountered while saving the member: %s."
 COM_USERS_USER_SAVE_SUCCESS="User saved."
+COM_USERS_USER_SEND_BUTTONTEXT="Activate and send email"
+COM_USERS_USER_SENDING_BUTTONTEXT="Sending..."
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.com_users.ini]: Add a space before `...`

File: administrator/language/en-GB/en-GB.com_users.ini [hunk 10]
```diff
@@ -79,12 +79,16 @@ COM_USERS_DESIRED_PASSWORD="Enter your desired password."
 COM_USERS_EDIT_NOTE="Edit Note"
 COM_USERS_EDIT_NOTE_N="Editing note with ID #%d"
 COM_USERS_EDIT_USER="Edit User %s"
+COM_USERS_EMAIL_ACTIVATED_BY_ADMIN_ACTIVATION_BODY="Hello %s,\n\nYour account has been activated by an administrator. You can now login at %s using the username %s and the password you chose while registering."
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.com_users.ini]: Not used.

File: administrator/components/com_users/views/user/tmpl/edit.php [hunk 11]
```diff
@@ -41,6 +43,65 @@
 	};
 ");
 
+// Perform a ajax request to active and send a email
```
Comment (Quy) [inline administrator/components/com_users/views/user/tmpl/edit.php]: Change `a` to `an` in both instances

**Comment by carlitorweb** [comment]
Thank again @Quy

File: administrator/components/com_users/models/user.php [hunk 12]
```diff
@@ -586,6 +588,58 @@ public function activate(&$pks)
 							return false;
 						}
 
+						if (!empty($wasActive))
+						{
+							// Then we wanted active the user and send a email notification
```
Comment (Quy) [inline administrator/components/com_users/models/user.php]: Change `a` to `an`.

File: administrator/components/com_users/models/user.php [hunk 13]
```diff
@@ -586,6 +588,58 @@ public function activate(&$pks)
...
+							// Compute the user mail notification subject.
+							$emailSubject = JText::sprintf(
+								'COM_USERS_EMAIL_ACTIVATED_BY_ADMIN_ACTIVATION_SUBJECT',
+								$table->name,
+								$config->get('sitename')
+							);
+
+							// Compute the user mail notification body.
+							$emailBody = JText::sprintf(
+								'COM_USERS_EMAIL_NOTIFICATION_ACTIVATED_BODY',
```
Comment (Quy) [inline administrator/components/com_users/models/user.php]: No language string.

File: administrator/components/com_users/models/user.php (hunk 12 above)
Comment (carlitorweb) [reply administrator/components/com_users/models/user.php]: ....an email notification.... this you mean?

File: administrator/components/com_users/models/user.php [hunk 14]
```diff
@@ -586,6 +588,58 @@ public function activate(&$pks)
...
+								$table->name,
+								JUri::root(),
+								$table->username
+							);
+						}
+						else
+						{
+							// Then we wanted only send again the email notification
```
Comment (Quy) [inline administrator/components/com_users/models/user.php]: Change to `Resend the email notification`?

File: administrator/components/com_users/controllers/user.php [hunk 15]
```diff
@@ -85,4 +85,54 @@ protected function postSaveHook(JModelLegacy $model, $validData = array())
...
+		$data  = $this->input->post->get('jform', array(), 'array');
+
+		if (empty($data['id']))
+		{
+			JError::raiseWarning(500, JText::_('COM_USERS_USERS_NO_ITEM_SELECTED'));
+		}
+		else
+		{
+			/* 
+			 *	Know if we call this for activate the user and send a email notification
```
Comment (Quy) [inline administrator/components/com_users/controllers/user.php]: Remove extra spaces after `*`. Change `a` to `an`.

File: administrator/components/com_users/controllers/user.php [hunk 16]
```diff
@@ -121,12 +121,12 @@ public function activate()
...
 				JError::raiseWarning(500, $model->getError());
 			}
-			// Then we wanted active the user and send a email notification
+			// Active the user and send an email notification
 			elseif (!empty($wasActive))
 			{
 				$this->setMessage(JText::sprintf('COM_USERS_USER_ACTIVATED_NOTIFIED', $data['name']));
 			}
-			// Then we wanted only send again the email notification
+			// Resend again the email notification
```
Comment (Quy) [inline administrator/components/com_users/controllers/user.php]: `Resend` implies again so `again` would be redundant.

File: administrator/components/com_users/controllers/user.php (hunk 15 above)
Comment (Quy) [reply administrator/components/com_users/controllers/user.php]: It should be like this `* Know`

File: administrator/components/com_users/controllers/activation.json.php [hunk 17]
```diff
@@ -0,0 +1,93 @@
...
+			$message = JText::_('COM_USERS_USERS_NO_ITEM_SELECTED');
+			$error = true;
+
+			return false;
+		}
+		else
+		{
+			$table = $model->getTable();
+			$table->load($userID);
+			$wasActive = $table->activation;
```
Comment (Quy) [inline administrator/components/com_users/controllers/activation.json.php]: `$wasActive` is unclear to me. Do you mean `$activated`?

Comment (carlitorweb) [reply administrator/components/com_users/controllers/activation.json.php]: Yes, as the model, active every time the user, I want to know before if an activation email was sent or not, to compute the message.

Comment (Quy) [reply administrator/components/com_users/controllers/activation.json.php]: Rename the variable name?

Comment (carlitorweb) [reply administrator/components/com_users/controllers/activation.json.php]: to $actived? for me both are good. No problem with rename to activated

File: administrator/components/com_users/controllers/user.php [hunk 18]
```diff
@@ -114,15 +114,15 @@ public function activate()
 			$model = $this->getModel();
 			$table = $model->getTable();
 			$table->load($data['id']);
-			$wasActive = $table->activation;
+			$actived = $table->activation;
```
Comment (Quy) [inline administrator/components/com_users/controllers/user.php:117]: `actived` is not a word. Do you mean `activate`?

**Review by mbabker — APPROVED** [review]

File: administrator/components/com_users/controllers/user.php (hunk 18 above)
Comment (infograf768) [reply administrator/components/com_users/controllers/user.php:117]: or rather `activated` I guess as the variable means the status of the user

Comment (carlitorweb) [reply administrator/components/com_users/controllers/user.php:117]: Yes, is how @infograf768 said. The status of the user before active he. (again or not).

**Comment by carlitorweb** [comment]
Some news about this?

File: administrator/components/com_users/controllers/activation.json.php [hunk 19]
```diff
@@ -0,0 +1,93 @@
...
+ * Activation controller class for manage the ajax request
+ *
+ * @since  __DEPLOY_VERSION__
+ */
+class UsersControllerActivation extends JControllerForm
+{
+	/**
+	 * Method to Active the user and send an email notification
+	 *
+	 * @param   string   &$message  The message that will be returned
```
Comment (Quy) [inline administrator/components/com_users/controllers/activation.json.php]: Remove `&` when in docblock.

**Comment by carlitorweb** [comment]
Thank @Quy

**Comment by twister65** [comment]
the build failed, just add & before parameter name (php doc):
* @param string &$message ...
https://github.com/joomla/joomla-cms/blob/6ee10ded413be3503e7140875919eec4db635cb4/administrator/components/com_users/controllers/activation.json.php#L22
* @param boolean &$error
https://github.com/joomla/joomla-cms/blob/6ee10ded413be3503e7140875919eec4db635cb4/administrator/components/com_users/controllers/activation.json.php#L23

**Comment by jjnxpct** [comment]
I have tested this item :white_check_mark: successfully on 6ee10ded413be3503e7140875919eec4db635cb4<br /><br />I tested this on Joomla 3.9. It seemed to work OK!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20282">issues.joomla.org/tracker/joomla-cms/20282</a>.</sub>

**Comment by jjnxpct** [comment]
I tested this patch. It seemed to work OK. 

If we want a site admin to use the backend (button) to activate users and not the the link in the mail (because this link now needs the user to be logged in, and sometimes a bakcend user does not have fornetnd backup) we could / should edit the email text and remove the activation link in a language override?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20282">issues.joomla.org/tracker/joomla-cms/20282</a>.</sub>

File: administrator/components/com_users/controllers/activation.json.php (hunk 19 above)
Comment (twister65) [reply administrator/components/com_users/controllers/activation.json.php]: The build failed : 
http://ci.joomla.org/joomla/joomla-cms/6505/4

    FILE: ...-cms/administrator/components/com_users/controllers/activation.json.php
    --------------------------------------------------------------------------------
    FOUND 2 ERROR(S) AFFECTING 2 LINE(S)
    --------------------------------------------------------------------------------
     22 | ERROR | Doc comment for var $message does not match actual variable name
        |       | &$message at position 1
     23 | ERROR | Doc comment for var $error does not match actual variable name
        |       | &$error at position 2

**Comment by twister65** [comment]
I have tested this item :white_check_mark: successfully on 6ee10ded413be3503e7140875919eec4db635cb4<br /><br />I think the button text : 
https://github.com/joomla/joomla-cms/blob/6ee10ded413be3503e7140875919eec4db635cb4/administrator/language/en-GB/en-GB.com_users.ini#L357
should be : 
    COM_USERS_USER_RESEND_BUTTONTEXT="Resend activation email"
because the user is already activated.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20282">issues.joomla.org/tracker/joomla-cms/20282</a>.</sub>

**Comment by jjnxpct** [comment]
@twister65 Not sure if that is right. They way I see this being used is that a new users creates an account, the verifies (not actovates) his/her email. Then the admin gets notified and can use this button to activate the user. Only then a notification email is send to the new user.

So this button would only be used with users that have not already been activated. 

Or will this also be used to resend this mail? Or are those actually two different functions?

I started looking at this PR after having issues with the link in the request-for-activation mail to the site manager did not work anymore without logging in to the (site (frontend). When adding this feature a site admin can - in stead of using the link in the mail - activate the user form the backend. Just by clicking on the activation icon in the user list. Or by clicking on the button.

But maybe I am missing something...

**Comment by jjnxpct** [comment]
Not sure if there is also a message that says 'Mail has been send to user' or something like that? To confirm to the sitemanager that activated the user that the mail was send out? Did not look for this while testing. Maybe I will have some time tomorrow to check this...

And what happens when an active user is deactivated in the backend and then reactivated again? Is the mail then send again? Do we want that to happen?

**Comment by twister65** [comment]
@jjnxpct, if the user is not activated the button text is different : 
https://github.com/joomla/joomla-cms/blob/6ee10ded413be3503e7140875919eec4db635cb4/administrator/language/en-GB/en-GB.com_users.ini#L360
I confirm that a message informs the site manager if the email has been sent or not.

**Comment by jjnxpct** [comment]
@twister65 ah, I see. We have 
COM_USERS_USER_SEND_BUTTONTEXT and 
COM_USERS_USER_RESEND_BUTTONTEXT

I am not quit sure in what situation you would need to resend this e-mail? Maybe the button could also just not show when already activated?

**Comment by jjnxpct** [comment]
What's next?

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20282">issues.joomla.org/tracker/joomla-cms/20282</a>.</sub>

**Comment by brianteeman** [comment]
What's preventing this important improvement from being merged?  @HLeithner

**Comment by HLeithner** [comment]
For my first thought it looks as a new feature for me and I don't know why it wasn't merged into 3.9 maybe it wasn't finished.

I would like to see it changed to J4. btw. there are some cs errors in the code.

**Comment by brianteeman** [comment]
It would be really great if someone could port this to joomla 4

**Comment by Quy** [comment]
Removed RTC. To be rebased for J4.

**Comment by ashvini77** [comment]
Error "The file marked for modification does not exist: administrator/components/com_users/controllers/user.php" displayed on applying the patch<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20282">issues.joomla.org/tracker/joomla-cms/20282</a>.</sub>

**Comment by brianteeman** [comment]
@ashvini77 This PR is for **Joomla 3** were you testing on Joomla 4?

**Comment by zero-24** [comment]
Dear @carlitorweb 

in preperation of the upcomming release of Joomla 3.10 we have used GitHubs rename feature to rename the staging branch into 3.10-dev. Usually GitHub moves all existing PRs towards the new branch just fine, but here it didnt work. The reason seems to be that the fork of the CMS that was used as base for this PR has been deleted so GitHub does no longer have a base to rebase the PR against the new branch and we are also not able to reopen the PR. For that reason GitHub closed this PR in my name, when this issue is still valid It would require a new PR against the new 3.10-dev or 4.0-dev branch.

**Comment by carlitorweb** [comment]
@zero-24 Understood. And is correct I deleted that fork. If this PR is still considered, what should I do?

**Comment by zero-24** [comment]
When this is still a feature you would like to have please create a PR against 4.1-dev so it can be tested and @bembelimen can take it into 4.1.

**Comment by jjnxpct** [comment]
Is this feature still being worked on? I have I site where this 'send registration email' button could be very useful ;-)

**Comment by mgretz** [comment]
We would also appreciate a working solution for this.

**Comment by brianteeman** [comment]
I am using this plugin now https://github.com/Spudley/plg_user_notifyactivation

**Comment by tschombes** [comment]
Still an issue in Joomla 5.x<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20282">issues.joomla.org/tracker/joomla-cms/20282</a>.</sub>

**Pull request closed without merging**